airflow/decorators/__init__.py

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import warnings
+from typing import Callable, Optional
+
+from airflow.decorators.python import python_task
+from airflow.models.dag import dag  # noqa # pylint: disable=unused-import
+
+
+class _TaskDecorator:
+    def __call__(
+        self, python_callable: Optional[Callable] = None, multiple_outputs: Optional[bool] = None, **kwargs
+    ):
+        """
+        Deprecated function that calls @task.python and allows users to turn a python function into
+        an Airflow task.
+        @param python_callable:
+        @param multiple_outputs:
+        @param kwargs:
+        @return:
+        """
+        warnings.warn(
+            """Calling airflow.decorators.task for python tasks is deprecated.
+            Please use @task.python instead.
```

Comment:
I don't like this really -- having it `@task.python` looses some of the "elegance" of the TaskFlow API.

Reply:
Also what about `@dag.task`? (I don't know how that's implemented, it may already delegate to this?)